docs/developers/connectors/architecture.md

```diff
@@ -134,12 +130,21 @@ Unlike `OrderBookTrackerDataSource`, `UserStreamTrackerDataSource` only retrieve
 
 ### InFlightOrder
 
-**File:** `*_in_flight_order.py` — Required
+**File:** `/hummingbot/core/data_type/in_flight_order.py`
+
+Stores all details pertaining to the current state of an order.
+
+It is important to keep a consistent and accurate state of all active orders placed by the user. This ensures that the strategies are given the correct information and are able to perform their tasks accordingly.
+
+### ClientOrderTracker
+
+**File:** `/hummingbot/connector/client_order_tracker.py`
+
+An instance of `ClientOrderTracker` holds and manages `InFlightOrders` for its connector.
+
+Provides utilities for connectors to update in-flight orders and to handle order errors.
```

Comment:
Although it is not really accurate to say this, do you think we can include triggering events as one of the responsibilities of the `ClientOrderTracker`?

Reason why I'm saying its not really accurate because technically, `ClientOrderTracker` just calls the `trigger_event` in the connector.

Reply:
I've updated the description to better reflect how it works.